libgraphics/include/pivot/graphics/types/AllocatedBuffer.hxx

```diff
@@ -3,65 +3,37 @@
 #include <vk_mem_alloc.hpp>
 #include <vulkan/vulkan.hpp>
 
-#include "pivot/graphics/VulkanBase.hxx"
-#include "pivot/graphics/vk_utils.hxx"
-
 namespace pivot::graphics
 {
 
-class AllocatedImage;
-
 /// @class AllocatedBuffer
 ///
 /// @brief Utility class to keep track of a Vulkan buffer and its allocated memory
 class AllocatedBuffer
 {
 public:
-    /// Constructor
-    AllocatedBuffer();
-    /// Destructor
-    ~AllocatedBuffer();
-
-    /// Clone the buffer into a new one
-    AllocatedBuffer cloneBuffer(VulkanBase &i, const vk::BufferUsageFlags usage, const vma::MemoryUsage memoryUsage);
-
-    template <vk_utils::is_copyable T>
-    /// Copy the data into a buffer
-    void copyBuffer(vma::Allocator &allocator, const T *data, size_t size)
-    {
-        void *mapped = allocator.mapMemory(memory);
-        std::memcpy(mapped, data, size);
-        allocator.unmapMemory(memory);
-    }
-
-    template <vk_utils::is_copyable T>
-    /// Copy the vector into the buffer
-    void copyBuffer(vma::Allocator &allocator, const std::vector<T> &data)
+    /// get the size of the buffer
+    auto getSize() const noexcept { return size; }
+    /// get the allocated size (can be different from the size)
+    auto getAllocatedSize() const noexcept { return info.size; }
+
+    template <typename T>
+    /// If the buffer was created with the flag vma::AllocationCreateFlagBits::eMapped, return the mapped pointer
```

Comment:
Please add an assertion to check that the buffer was indeed created with this flag.